src/splitseq_run.sh

```diff
@@ -14,6 +14,17 @@ ssrefdir=$5
 raw_samples=("${all_args[@]:5}")
 
 
+#Gzip the inputs if not gzipped (a splitseq requirement)
+if [ ${mrna_fq: -3} != ".gz" ]; then
```

Comment:
This will obviously work, though a more robust way to get the extension would be using [bash parameter expansion](https://wiki.bash-hackers.org/syntax/pe), *e.g.*:
```bash
extension="${filename##*.}"
```

Reply:
I'm not very bash savvy, obviously, but `${mrna_fq: -3} != ".gz"` seems easier to read to me.  What does `##*.` do?  If I were to guess (without looking it up), I'd venture that `##*` is a greedy pattern substitution (or is it only up to the first dot)?

Reply:
No doubt it's simpler, but always pulls the last three characters (fine in this instance). But to get the extension, check out https://wiki.bash-hackers.org/syntax/pe#substring_removal

One `#` removes up to the first dot. `##` is greedy.

Reply:
Definitely bookmarking that resource.  Thanks.